src/lib/table/table.md

```diff
@@ -1,50 +1,185 @@
-with `mat-` instead of `cdk-`.
-
-For more information on the interface and how it works, see the
+with `mat-` instead of `cdk-`. For detailed information on the interface and how it works, see the 
 [guide covering the CDK data-table](https://material.angular.io/guide/cdk-table).
 
+### Getting Started
+
 <!-- example(table-basic) -->
 
-### Features
+Start by writing your table's column definitions. Each column definition should be given a unique 
+name and contain the content definitions for its header and row cells.
+
+Here's a simple column definition with the name `'userName'`. The header cell contains the text 
+"Name" and each row cell will render the `name` property of each row's data.
+
+```html
+<ng-container matColumnDef="userName">
+  <mat-header-cell *matHeaderCellDef> Name </mat-header-cell>
+  <mat-cell *matCellDef="let user"> {{user.name}} </mat-cell>
+</ng-container>
+```
+
+After you define all your columns, you'll need to provide the header and data row templates that 
+will be rendered out by the table. Each template should be given an ordered list of columns that 
+will let the table know what columns each row should render.
+
+```html
+<mat-header-row *matHeaderRowDef="['userName', 'age']"></mat-header-row>
+<mat-row *matRowDef="let myRowData; columns: ['userName', 'age']"></mat-row>
+```
+
+Once your template is set up, the final step is to provide data to the table. This must be done 
+through a `DataSource`, which uses an Observable stream that emits an array of data that the table 
+should render. 
+
+To get started, you can use the pre-built `MatTableDataSource` to handle the stream for you. 
+Just create a new instance and set its `data` property to whatever data you want the table to 
+render.
 
-The `<mat-table>` itself only deals with the rendering of a table structure (rows and cells).
-Additional features can be built on top of the table by adding behavior inside cell templates
-(e.g., sort headers) or next to the table (e.g. a paginator). Interactions that affect the
-rendered data (such as sorting and pagination) should be propagated through the table's data source.
+```ts
+this.myDataSource = new MatTableDataSource();
+this.myDataSource.data = dataToRender;
+```
 
+```html
+<mat-table [dataSource]=”myDataSource”>
+  ...
+</mat-table>
+```
+
+### Features
 
 #### Pagination
 
-The `<mat-paginator>` adds a pagination UI that can be used in conjunction with the `<mat-table>`. The
-paginator emits events that can be used to trigger an update via the table's data source.
+To paginate the table's data, add a `<mat-paginator>` after the `<mat-table>` and provide the 
+`MatPaginator` to the `MatTableDataSource`. The data source will automatically listen for page 
+changes made by the user and send the right paged data to the table.
+
+For more information on using and configuring the `<mat-paginator>`, check out the 
+[mat-paginator docs](https://material.angular.io/components/paginator/overview).
```

Comment:
We should mention here that `mat-paginator` is just one option. The table can work with any paginator control or any pagination strategy.

(same for sorting, filtering, and selection)